AutomatticTracks/src/main/java/com/automattic/android/tracks/DeviceInformation.java

```diff
@@ -242,14 +242,18 @@ public Boolean isWifiConnected() {
     }
 
     public Boolean isBluetoothEnabled() {
+        // Do not retrieve bluetooth info on older Android versions.
+        // See: https://github.com/Automattic/Automattic-Tracks-Android/issues/20
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
+            return null;
+        }
+
         Boolean isBluetoothEnabled = null;
-        try {
+        if (PackageManager.PERMISSION_GRANTED == mContext.checkCallingOrSelfPermission(Manifest.permission.BLUETOOTH)) {
```

Comment:
We're already using `checkCallingOrSelfPermission` in other part of the lib without issues. It's not opening a security issue and offers the ability to access the library in IPC cases (that's not our case atm).

 I went beyond the doc since it's not so clear on this, and found out that in our cases (Tracks and WPAndroid) the call is identical to `checkPermission `. If we open the library/app to IPC we will enable the caller to access the BT info (if our process has the ability to access BT). We can change it to `checkPermission` safely though.